### DISCUSSION TITLE
Skip partition materialization at runtime

### DISCUSSION THREAD

**Discussion by jeanineharb** [dagster-io/dagster#18666] [Q&A] [answered]
I'm implementing a job with several assets partitioned by department key (static partitions).
Some of these assets can't find the data related to one or two departments out of 100 : source data is not available openly.

Is there a way to skip the materialization of a static partition without marking the run as failed ? (when the data is not available at the source)

**Comment by sryza** [comment, marked as answer]
Hi @jeanineharb  - you can use `output_required` for this:

```python
@asset(partitions_def=..., output_required=False)
def myasset():
    if data_is_available():
        yield Output()
```

**Reply by jeanineharb** [reply]
Thank you for your suggestion! Works great :)
